# Excercises Electric Machinery Fundamentals
## Chapter 4

## Problem 4-3

In [1]:
%pylab inline

Populating the interactive namespace from numpy and matplotlib


### Description

Assume that the field current of the generator in Problem 4-2 has been adjusted to a value of $5\,A$.

#### (a)
 
 * What will the terminal voltage of this generator be if it is connected to a $\Delta$-connected load with an impedance of $24\,\Omega\angle 25°$?
 
#### (b)
 
 * Sketch the phasor diagram of this generator.

#### (c)
 
 * What is the efficiency of the generator at these conditions?
 
#### (d)
Now assume that another identical $\Delta$-connected load is to be paralleled with the first one. 

 * What happens to the phasor diagram for the generator?
 
#### (e)
 
 * What is the new terminal voltage after the load has been added?
 
#### (f)

 * What must be done to restore the terminal voltage to its original value?

In [2]:
If = 5.0 # [A]
PF = 0.9
Xs = 2.5 # [Ohm]
Ra = 0.2 # [Ohm]
Zload = 24 * (cos(25/180.0 * pi) + sin(25/180.0 * pi)*1j)
P      =  50e6 # [W]
Pf_w   = 1.0e6 # [W]
Pcore  = 1.5e6 # [W]
Pstray =     0 # [W]
n_m = 1800     # [r/min]

### SOLUTION

#### (a)
If the field current is $5.0 A$, the open-circuit terminal voltage will be about $16,500\,V$, and the open-circuit phase voltage in the generator (and hence $E_A$ ) will be $\frac{16,500\,V}{\sqrt{3}}$ .

In [3]:
Vl = 16.5e3 #[V]
ia = P / (sqrt(3) * Vl)
Ia_angle = -arccos(PF)
Ia = ia * (cos(Ia_angle) + sin(Ia_angle)*1j)

Ea = Vl / sqrt(3)
print('Ea = {:.0f} V'.format(Ea))

Ea = 9526 V


The load is $\Delta$-connected with three impedances of $24\,\Omega \angle 25^\circ$ . From the Y-$\Delta$ transform, this load is equivalent to a Y-connected load with three impedances of:

In [4]:
Z = Zload/3
Z_angle = arctan(Z.imag/Z.real)
print('Z = {:.0f} Ω ∠{:.0f}° '.format(abs(Z), Z_angle / pi*180))

Z = 8 Ω ∠25° 


The resulting per-phase equivalent circuit is shown below:

<img src="figs/Problem_4-03a.jpg" width="70%">

The magnitude of the phase current flowing in this generator is:

$$I_A = \frac{E_A}{|R_A + jX_S +Z|}$$

In [5]:
ia = Ea / (abs(Ra + Xs*1j + Z))
print('ia = {:.0f} A'.format(ia))
Ia = ia * (cos(-Z_angle) + sin(-Z_angle)*1j)
Ia_angle = arctan(Ia.imag/Ia.real)

ia = 1004 A


Therefore, the magnitude of the phase voltage is:
$$V_\phi = I_AZ$$

In [6]:
V_phase = ia * abs(Z)
print('V_phase = {:.0f} V'.format(V_phase))

V_phase = 8029 V


and the terminal voltage is:
$$V_T = \sqrt{3}V_\phi$$

In [7]:
Vt = sqrt(3) * V_phase
print('''
Vt = {:.0f} V
============'''.format(Vt))


Vt = 13907 V


#### (b)

Armature current is $I_A = 1004\,A\angle -25°$ , and the phase voltage is $V_\phi = 8032\,V\angle 0°$. Therefore, the internal generated voltage is:

$$\vec{E}_A = \vec{V}_\phi + R_A\vec{I}_A + jX_S\vec{I}_A$$

In [8]:
EA = V_phase + Ra*Ia + Xs*1j*Ia
EA_angle = arctan(EA.imag/EA.real)
print('EA = {:.0f} V ∠{:.1f}°'.format(abs(EA), EA_angle / pi *180))

EA = 9526 V ∠13.3°


The resulting phasor diagram is shown below (not to scale and with some round-off errors):

<img src="figs/Problem_4-03b.jpg" width="70%">

#### (c)
The efficiency of the generator under these conditions can be found as follows:

$$P_\text{out} = 3 V_\phi I_A \cos{\theta}$$

In [9]:
Pout = 3 * V_phase * abs(Ia) * cos(Ia_angle)
print('Pout = {:.1f} MW'.format(Pout/1e6))

Pout = 21.9 MW


$$P_\text{CU} = 3I^2_AR_A$$

In [10]:
Pcu = 3 * abs(Ia)**2 * Ra
print('Pcu = {:.1f} kW'.format(Pcu/1e3))

Pcu = 604.4 kW


$P_\text{F\&W} = 1\,MW$:

In [11]:
Pf_w = 1e6 # [W] 

$P_\text{core} = 1.5\,MW$:

In [12]:
Pcore = 1.5e6 # [W]

$P_\text{stray} \approx 0$ (assumed):

In [13]:
Pstray = 0 # [W]

$$P_\text{in} = P_\text{out} + P_\text{CU} + P_\text{F\&W} + P_\text{core} + P_\text{stray}$$

In [14]:
Pin = Pout + Pcu + Pf_w + Pcore + Pstray
print('Pin = {:.1f} MW'.format(Pin/1e6))

Pin = 25.0 MW


$$\eta = \frac{P_\text{out}}{P_\text{in}} \cdot 100\%$$

In [15]:
eta = Pout / Pin
print('''
η = {:.1f} %
=========='''.format(eta*100))


η = 87.6 %


#### (d)
To get the basic idea of what happens, we will ignore the armature resistance for the moment. If the field current and the rotational speed of the generator are constant, then the magnitude of $E_A( = K \phi\omega)$ is constant. The quantity $jX_S \vec{I}_A$ increases in length at the same angle, while the magnitude of $\vec{E}_A$ must remain constant. Therefore, $\vec{E}_A$ “swings” out along the arc of constant magnitude until the new $jX_S \vec{I}_S$ fits exactly between $\vec{V}_\phi$ and $\vec{E}_A$ .

<img src="figs/Problem_4-03c.jpg" width="60%">

#### (e)
The new impedance per phase will be half of the old value, so 

In [16]:
Znew = Z * 0.5
Znew_angle = arctan(Z.imag/Z.real)
print('Z = {:.0f} Ω ∠{:.0f}° '.format(abs(Znew), Znew_angle /pi*180))

Z = 4 Ω ∠25° 


The magnitude of the phase current flowing in this generator is:
$$I_A = \frac{E_A}{|R_A + jX_S +Z_\text{new}|}$$

In [17]:
ia = Ea / (abs(Ra + Xs*1j + Znew))
print('ia = {:.1f} A'.format(ia))

ia = 1679.0 A


Therefore, the magnitude of the phase voltage is:
$$V_\phi = I_AZ_\text{new}$$

In [18]:
V_phase = ia * abs(Znew)
print('V_phase = {:.1f} V'.format(V_phase))

V_phase = 6715.9 V


and the terminal voltage is:

$$V_T = \sqrt{3}V_\phi$$

In [19]:
Vt = sqrt(3) * V_phase
print('''
Vt = {:.1f} V
=============='''.format(Vt))


Vt = 11632.3 V


#### (f)
To restore the terminal voltage to its original value, increase the field current: $I_F\uparrow$ .